### PR TITLE
Allow passing a custom domain and max domain length to pullpreview. 

### PR DESCRIPTION
**Pull request by holyfabi**

### CODE REVIEW

File: .github/workflows/pullpreview.yml [hunk 1]
```diff
@@ -17,6 +17,14 @@ on:
         description: The name of an Artifact which will be extracted while preserving file paths.
         required: false
         type: string
+      DNS:
```
Comment (mszulik) [inline .github/workflows/pullpreview.yml:20]:
```suggestion
      CUSTOM_DOMAIN:
```

File: .github/workflows/pullpreview.yml [hunk 2]
```diff
@@ -17,6 +17,14 @@ on:
...
+        description: A custom domain which will be used for the preview environment.
+        required: false
+        type: string
+      MAX_DOMAIN_LENGTH:
+        description: The maximum length which will be used for the fully qualified domain name.
```
Comment (mszulik) [inline .github/workflows/pullpreview.yml:25]: Maybe add here or below what exactly this does. Like, it cuts off the subdomain name to fit into this max length.

File: .github/workflows/pullpreview.yml [hunk 3]
```diff
@@ -96,6 +104,10 @@ jobs:
           instance_type: ${{ inputs.INSTANCE_TYPE }}
           # Ports to open on the server
           ports: 80, 443
+          # Use a custom domain for the preview environment
+          dns: ${{ inputs.DNS }}
```
Comment (mszulik) [inline .github/workflows/pullpreview.yml:108]: what happens if the input is empty?

File: .github/workflows/pullpreview.yml [hunk 4]
```diff
@@ -96,6 +104,10 @@ jobs:
...
+          # Maximum length of fully qualified domain name. Cannot be greater than 62 characters.
+          max_domain_length: ${{ inputs.MAX_DOMAIN_LENGTH }}
```
Comment (mszulik) [inline .github/workflows/pullpreview.yml:110]: add the new keys to the example and comment them out.

File: .github/workflows/pullpreview.yml (hunk 1 above)
Comment (mszulik) [reply .github/workflows/pullpreview.yml:20]: or `DOMAIN`

Comment (holyfabi) [reply .github/workflows/pullpreview.yml]: Changed it to `DOMAIN` like suggested. Done.

File: .github/workflows/pullpreview.yml (hunk 2 above)
Comment (holyfabi) [reply .github/workflows/pullpreview.yml]: Done.

File: .github/workflows/pullpreview.yml (hunk 4 above)
Comment (holyfabi) [reply .github/workflows/pullpreview.yml:113]: Done.

File: .github/workflows/pullpreview.yml (hunk 3 above)
Comment (holyfabi) [reply .github/workflows/pullpreview.yml]: PullPreview leaves the domain empty. The url therefore only consists of the protocol and the subdomain. 

Therefore there is now an fallback to the default PullPreview url, when the input is empty.

File: .github/workflows/pullpreview.yml [hunk 5]
```diff
@@ -96,6 +106,11 @@ jobs:
           instance_type: ${{ inputs.INSTANCE_TYPE }}
           # Ports to open on the server
           ports: 80, 443
+          # It seems, although not identical, <condition> && <true-value> || <false-value> works in most cases.
```
Comment (mszulik) [inline .github/workflows/pullpreview.yml:109]: Identical to what? What do you mean with most cases?

Comment (holyfabi) [reply .github/workflows/pullpreview.yml]: This comment wasn't supposed to be there. Let's just pretend you didn't see anything. Removed the Line...

**Review by mszulik — APPROVED** [review]
Looks good, should be tested somewhere though.

**Review by gael-connan-cybex — APPROVED** [review]
proven to work in https://github.com/cybex-gmbh/redirector/pull/15

**Pull request merged**